internal/buffercloser/buffercloser.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The ebml-go authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buffercloser
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"sync"
+)
+
+// ErrClosedBuffer is the error returned by Write on closed buffer.
+var ErrClosedBuffer = errors.New("write on closed buffer")
+
+// BufferCloser is bytes.Buffer with io.Closer interface.
+type BufferCloser interface {
+	io.Writer
+	io.Closer
+	Bytes() []byte
+	Closed() <-chan struct{}
+}
+
+type bufferCloser struct {
+	buf       bytes.Buffer
+	closed    chan struct{}
+	closeOnce sync.Once
+}
+
+func New() BufferCloser {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
exported function New should have comment or be unexported